sql/ddl_api.sql

```diff
@@ -194,50 +135,15 @@ $BODY$;
 --     microseconds, or an INTERVAL type.
 CREATE OR REPLACE FUNCTION  set_chunk_time_interval(
     main_table              REGCLASS,
-    chunk_time_interval     anyelement
-)
-    RETURNS VOID LANGUAGE PLPGSQL VOLATILE
-    SECURITY DEFINER SET search_path='_timescaledb_catalog'
-    AS
-$BODY$
-DECLARE
-    main_table_name            NAME;
-    main_schema_name           NAME;
-    chunk_time_interval_actual BIGINT;
-    time_dimension             _timescaledb_catalog.dimension;
-    time_type                  REGTYPE;
-BEGIN
-    PERFORM _timescaledb_internal.check_role(main_table);
-    IF chunk_time_interval IS NULL THEN
-        RAISE EXCEPTION 'chunk_time_interval cannot be NULL'
-        USING ERRCODE = 'IO102';
-    END IF;
-
-    SELECT relname, nspname
-    INTO STRICT main_table_name, main_schema_name
-    FROM pg_class c
-    INNER JOIN pg_namespace n ON (n.OID = c.relnamespace)
-    WHERE c.OID = main_table;
+    chunk_time_interval     ANYELEMENT,
+    dimension_name          NAME = NULL
+) RETURNS VOID AS '@MODULE_PATHNAME@', 'dimension_set_interval' LANGUAGE C VOLATILE;
 
-    SELECT *
-    INTO STRICT time_dimension
-    FROM _timescaledb_internal.dimension_get_time(
-        (
-            SELECT id
-            FROM _timescaledb_catalog.hypertable h
-            WHERE h.schema_name = main_schema_name AND
-            h.table_name = main_table_name
-    ));
-
-    time_type := _timescaledb_internal.dimension_type(main_table, time_dimension.column_name, true);
-    chunk_time_interval_actual := _timescaledb_internal.time_interval_specification_to_internal_with_default_time(
-        time_type, chunk_time_interval, 'chunk_time_interval');
-
-    UPDATE _timescaledb_catalog.dimension d
-    SET interval_length = chunk_time_interval_actual
-    WHERE time_dimension.id = d.id;
-END
-$BODY$;
+CREATE OR REPLACE FUNCTION  set_num_partitions(
```

Comment:
Should this be `set_number_partitions` to be the same as the parameter name in create_hypertable?

Reply:
I think that name sounds somewhat awkward, but can change.